index.bs

```diff
@@ -1252,6 +1252,48 @@ See also:
 
 * [[#attributes-like-data]]
 
+<h3 id="attributes-vs-methods">Use attributes or methods adequately</h3>
+
+Sometimes it is unclear
+whether to use an attribute or a method.
+
+- Attribute getters must not have any (observable) side effects.
+    If you have expected side effects, use a method.
+
+- Attribute getters should not perform any expensive operations.
+    If a getter requires performing an expensive operation,
+    it should be a method.
+    (A notable failure of the platform in this regard
+    is getters like offsetTop performing layout;
+    do not repeat this mistake.)
```

Comment:
Do you have an example of the latter?

Reply:
I would like an example too. I think when conversion or involved it would be better as a method, like json(), toJSON(), toBlob(). Here I would expect an attribute to just return a value and not really do any calculation

Reply:
How about considering that blocking for some time is a kind of side-effect (and then see the point above)